src/test/java/uk/ac/cam/cl/dtg/isaac/api/QuizFacadeIT.java

```diff
@@ -110,5 +114,137 @@ TEST_TUTOR_ID, TEST_TUTORS_AB_GROUP_ID, new Date(), DateUtils.addDays(new Date()
+            NoSuchAlgorithmException, MFARequiredButNotConfiguredException {
+        // Arrange
+        // log in as Tutor, create request
+        LoginResult teacherLogin = loginAs(httpSession, TEST_TUTOR_EMAIL, TEST_TUTOR_PASSWORD);
+        HttpServletRequest assignQuizRequest = createRequestWithCookies(new Cookie[]{teacherLogin.cookie});
+        replay(assignQuizRequest);
+
+        // Act
+        // make request
+        Response getQuizzesResponse = quizFacade.getAvailableQuizzes(createNiceMock(Request.class), assignQuizRequest);
+
+        // Assert
+        // check status code is OK
+        assertEquals(Response.Status.OK.getStatusCode(), getQuizzesResponse.getStatus());
+
+        // check invisible-to-student and hidden-from-tutor-role quizzes are not returned as available
+        @SuppressWarnings("unchecked") ResultsWrapper<QuizSummaryDTO> responseBody =
+                (ResultsWrapper<QuizSummaryDTO>) getQuizzesResponse.getEntity();
+        assertTrue(responseBody.getResults().stream().anyMatch(q -> q.getId().equals(QUIZ_TEST_QUIZ_ID)));
+        assertFalse(responseBody.getResults().stream().anyMatch(q -> q.getId().equals(QUIZ_INVISIBLE_TO_STUDENTS_QUIZ_ID)));
+        assertFalse(responseBody.getResults().stream().anyMatch(q -> q.getId().equals(QUIZ_HIDDEN_FROM_ROLE_TUTORS_QUIZ_ID)));
+    }
+
+    @Test
+    public void previewQuizEndpoint_previewInvisibleToStudentQuizAsTeacher_succeeds() throws NoCredentialsAvailableException,
+            NoUserException, SegueDatabaseException, AuthenticationProviderMappingException,
+            IncorrectCredentialsProvidedException, AdditionalAuthenticationRequiredException, InvalidKeySpecException,
+            NoSuchAlgorithmException, MFARequiredButNotConfiguredException {
+        // Arrange
+        // log in as Teacher, create request
+        LoginResult teacherLogin = loginAs(httpSession, TEST_TEACHER_EMAIL, TEST_TEACHER_PASSWORD);
+        HttpServletRequest assignQuizRequest = createRequestWithCookies(new Cookie[]{teacherLogin.cookie});
+        replay(assignQuizRequest);
+
+        // Act
+        // make request
+        Response previewQuizResponse = quizFacade.previewQuiz(createNiceMock(Request.class), assignQuizRequest,
+                QUIZ_INVISIBLE_TO_STUDENTS_QUIZ_ID);
+
+        // Assert
+        // check status code is OK
+        assertEquals(Response.Status.OK.getStatusCode(), previewQuizResponse.getStatus());
+
+        // check the quiz is returned for preview
+        IsaacQuizDTO responseBody = (IsaacQuizDTO) previewQuizResponse.getEntity();
+        assertEquals(QUIZ_INVISIBLE_TO_STUDENTS_QUIZ_ID, responseBody.getId());
+    }
+
+    @Test
+    @Ignore("Skipping as this fails - we should decide what the expectations are here.")
+    public void previewQuizEndpoint_previewInvisibleToStudentQuizAsTutor_fails() throws NoCredentialsAvailableException,
```

Comment:
As the comment says, I'm not sure if quizzes where `visibleToStudents: false` should be preview-able by tutors.

Reply:
I think that's an edge case I missed, cheers for spotting that 🙌 Those definitely shouldn't be preview-able since it goes right against the semantics of the `visibleToStudents` flag.

Reply:
I'll fix this (shouldn't need any front-end since tutors only see the student-only tests anyway, but I'll double check that this is the case)